src/Microsoft.Azure.SignalR/HubHost/ServiceHubDispatcher.cs

```diff
@@ -38,12 +39,14 @@ internal class ServiceHubDispatcher<THub> where THub : Hub
             ILoggerFactory loggerFactory,
             IEndpointRouter router,
             IServerNameProvider nameProvider,
-            IClientConnectionFactory clientConnectionFactory)
+            IClientConnectionFactory clientConnectionFactory,
+            ServiceScaleManager serviceScaleManager)
         {
             _serviceProtocol = serviceProtocol;
             _serviceConnectionManager = serviceConnectionManager;
             _clientConnectionManager = clientConnectionManager;
             _serviceEndpointManager = serviceEndpointManager;
+            _serviceScaleManager = serviceScaleManager;
```

Comment:
if `ServiceScaleManager ` is a dependency, use interface? 
https://stackoverflow.com/questions/43079277/do-we-need-interfaces-for-dependency-injection
> Making your consumer classes depend on interfaces (abstractions) instead of other concrete classes makes them much easier to test.

Reply:
Sure. The interface is added in later PRs.

---
In reply to: [374094131](https://github.com/Azure/azure-signalr/pull/801#discussion_r374094131) [](ancestors = 374094131)